default/AI/AIstate.py

```diff
@@ -120,8 +127,28 @@ def __setstate__(self, state_dict):
         for odict_attrib in ['colonisablePlanetIDs', 'colonisableOutpostIDs']:
             if dict_attrib not in state_dict:
                 self.__dict__[odict_attrib] = odict()
+        if 'uuid' not in state_dict:
+            self.uuid = uuid4()
+        if 'turn_hashes' not in state_dict:
+            self.turn_uuids = {}
         self.__dict__.setdefault('empire_standard_enemy_rating', 40)
 
+    def set_turn_uuid(self):
+        """
+        Set turn uuid. Should be called once per generateOrders
+        New uuid set for each generate orders.
+        In case of load it can be executed multiple time during turn.
+        """
+        uuid = uuid4()
+        self.turn_uuids[fo.currentTurn()] = uuid
+        return uuid
+
+    def get_prev_turn_uuid(self):
+        """
+        Return hash of previous turn. In case of old save games return None
```

Comment:
"In case of old save games" would often not be accurate-- it looks to me like it would be better to say something like  "If called during the first turn after loading a saved game that had an AI version not yet using uuids,"